src/services/test-helpers/mock/assets/mockAssetData.ts

```diff
@@ -0,0 +1,157 @@
+import { PalletAssetsAssetAccount } from '@polkadot/types/lookup';
+import { Option } from '@polkadot/types-codec/base';
+
+import { statemintV1 } from '../../../../test-helpers/metadata/statemintMetadata';
+import { rococoRegistry } from '../../../../test-helpers/registries';
+import { createApiWithAugmentations } from '../../../../test-helpers/typeFactory';
+import { TypeFactory } from '../../../../test-helpers/typeFactory';
+
+/**
+ * This mock data uses Statemint specVersion 1
+ */
+const statemintApiV1 = createApiWithAugmentations(statemintV1);
+const statemintTypeFactory = new TypeFactory(statemintApiV1);
+
+const falseBool = rococoRegistry.createType('bool', false);
+const trueBool = rococoRegistry.createType('bool', true);
+const assetTBalanceOne = rococoRegistry.createType('u64', 10000000);
+const assetTBalanceTwo = rococoRegistry.createType('u64', 20000000);
+
+const accountIdOne = rococoRegistry.createType(
+	'AccountId',
+	'1TYrFCWxwHA5bhiXf6uLvPfG6eEvrzzL7uiPK3Yc6yHLUqc'
+);
+const accountIdTwo = rococoRegistry.createType(
+	'AccountId',
+	'13NXiLYYzVEjXxU3eaZNcrjEX9vPyVDNNpURCzK8Bj9BiCWH'
+);
+const balanceOfTwo = rococoRegistry.createType('BalanceOf', 2000000);
+
+const assetBalanceObjOne = {
+	balance: assetTBalanceOne,
+	isFrozen: falseBool,
+	sufficient: trueBool,
+};
+
+const assetBalanceObjTwo = {
+	balance: assetTBalanceTwo,
+	isFrozen: trueBool,
+	sufficient: trueBool,
+};
+
+const assetBalanceObjThree = {
+	balance: assetTBalanceTwo,
+	isFrozen: falseBool,
+	sufficient: falseBool,
+};
+
+const assetBalanceFactory = {
+	'10': assetBalanceObjOne as unknown as PalletAssetsAssetAccount,
+	'20': assetBalanceObjTwo as unknown as PalletAssetsAssetAccount,
+	'30': assetBalanceObjThree as unknown as PalletAssetsAssetAccount,
+};
+
+const assetStorageKeyOne = statemintTypeFactory.storageKey(
+	10,
+	'AssetId',
+	statemintApiV1.query.assets.asset
+);
+
+const assetStorageKeyTwo = statemintTypeFactory.storageKey(
+	20,
+	'AssetId',
+	statemintApiV1.query.assets.asset
+);
+
+const assetStorageKeyThree = statemintTypeFactory.storageKey(
+	30,
+	'AssetId',
+	statemintApiV1.query.assets.asset
+);
+
+const assetsAccountKeysAt = () =>
+	Promise.resolve().then(() => {
+		return [assetStorageKeyOne, assetStorageKeyTwo, assetStorageKeyThree];
+	});
+
+export const assetsInfo = (): Promise<AssetDetails> =>
+	Promise.resolve().then(() => {
+		const responseObj = {
+			owner: accountIdOne,
+			issue: accountIdTwo,
+			admin: accountIdTwo,
+			freezer: accountIdTwo,
+			supply: assetTBalanceOne,
+			deposit: balanceOfTwo,
+			minBalance: rococoRegistry.createType('u64', 10000),
+			isSufficient: trueBool,
+			accounts: rococoRegistry.createType('u32', 10),
+			sufficients: rococoRegistry.createType('u32', 15),
+			approvals: rococoRegistry.createType('u32', 20),
+			isFrozen: falseBool,
+		};
+
+		return rococoRegistry.createType('AssetDetails', responseObj);
+	});
+
+export const assetsInfoKeysInjected = (): (() => Promise<AssetDetails>) => {
```

Comment:
To highlight a change for reviewers. I use to directly inject the `keys` field into the `assetsInfo` in order to replicate the stucture of the object polkadot-js. I  used Object.assign() directly in the top layer of the file without enclosing it within its own function. I removed that by adding the exported `assetsInfoKeysInjected`. This way we don't tamper with `assetsInfo` ever within testing, but instead can have the keys if it's necessary with a shallow copy of the original object with the keys injected.

Reply:
Cool, I like this change!

Since JS is so awful at preventing things from being mutated I do like the "safe" option of returning objects and such from functions for tests; that way you know nothing else can ever mess with them!

Reply:
😺 Yea I totally agree. Glad I changed that in the refactoring in this PR, very thankful for the TS compiler in moments like this as well. Makes it super easy to cleanup.